pkg/cmd/workspace/start.go

```diff
@@ -112,20 +113,31 @@ var StartCmd = &cobra.Command{
 
 		if startProjectFlag == "" {
 			message = fmt.Sprintf("Workspace '%s' is starting", workspaceIdOrName)
-			res, err := apiClient.WorkspaceAPI.StartWorkspace(ctx, workspaceIdOrName).Execute()
+			err = views_util.WithSpinner(message, func() error {
+				res, err := apiClient.WorkspaceAPI.StartWorkspace(ctx, workspaceIdOrName).Execute()
+				if err != nil {
+					return apiclient.HandleErrorResponse(res, err)
+				}
+				return nil
+			})
 			if err != nil {
-				log.Fatal(apiclient.HandleErrorResponse(res, err))
+				log.Fatal(err)
 			}
-
-			views.RenderInfoMessage(message)
+			views.RenderInfoMessage(fmt.Sprintf("Workspace '%s' started successfully", workspaceIdOrName))
 		} else {
 			message = fmt.Sprintf("Project '%s' from workspace '%s' is starting", startProjectFlag, workspaceIdOrName)
-			res, err := apiClient.WorkspaceAPI.StartProject(ctx, workspaceIdOrName, startProjectFlag).Execute()
+			err = views_util.WithSpinner(message, func() error {
```

Comment:
Since this code is duplicated in 5 places, I suggest you create a function that you can call in all 5 places. This will make the code more maintainable.

Reply:
done